core/src/main/java/org/mule/runtime/core/internal/util/TempFileUtils.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.util;
+
+import static java.lang.System.getProperty;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+
+import java.io.File;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Utility methods to handle temporal files
+ *
+ * @since 4.0
+ */
+public final class TempFileUtils {
```

Comment:
This class contains method that seems to be useful for other use cases than streaming, like createTempFile. That mehtod should go in FileUtils and for things more specific to streaming like createBufferFile, that should have an util within the streaming package.